### ISSUE TITLE
Stuck on splash screen

### ISSUE THREAD

**Issue by omichaelo** [blockbasti/just_another_workout_timer#36]
After launching the app, the app remains on the splash screen with the icon and remains nonfunctional. I find this problem only applies to versions after 1.1.1. 

I am more than willing to give more debugging information if someone can tell me how to obtain them. Thanks!

**Comment by blockbasti** [comment]
I rarely get this bug when developing on my emulator. I haven't found a way to reproduce this yet or how to fix it.

@mschmidm have you encountered this problem when developing?

**Comment by mschmidm** [comment]
This occurred to me when I was setting up the project, but I got around it by using a lower Android version. Everything works fine with Android 9 and 10, but with Android 11 I can reproduce this bug.
Emulator: Pixel 3a API 30
<details>
  <summary>Console Output</summary>

  ```
Launching lib/main.dart on sdk gphone x86 arm in debug mode...
Running Gradle task 'assembleDebug'...
Plugin project :soundpool_web not found. Please update settings.gradle.
Plugin project :soundpool_macos not found. Please update settings.gradle.
✓ Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...
Waiting for sdk gphone x86 arm to report its views...
Debug service listening on ws://127.0.0.1:32871/U6LJt3v1CZg=/ws
Syncing files to device sdk gphone x86 arm...
D/CCodec  ( 6336): allocate(c2.android.raw.decoder)
I/Codec2Client( 6336): Available Codec2 services: "software"
I/CCodec  ( 6336): Created component [c2.android.raw.decoder]
D/CCodecConfig( 6336): read media type: audio/raw
D/ReflectedParamUpdater( 6336): extent() != 1 for single value type: algo.buffers.max-count.values
D/ReflectedParamUpdater( 6336): extent() != 1 for single value type: output.subscribed-indices.values
D/ReflectedParamUpdater( 6336): extent() != 1 for single value type: input.buffers.allocator-ids.values
D/ReflectedParamUpdater( 6336): extent() != 1 for single value type: output.buffers.allocator-ids.values
D/ReflectedParamUpdater( 6336): extent() != 1 for single value type: algo.buffers.allocator-ids.values
D/ReflectedParamUpdater( 6336): extent() != 1 for single value type: output.buffers.pool-ids.values
D/ReflectedParamUpdater( 6336): extent() != 1 for single value type: algo.buffers.pool-ids.values
I/CCodecConfig( 6336): query failed after returning 8 values (BAD_INDEX)
D/CCodecConfig( 6336): c2 config diff is Dict {
D/CCodecConfig( 6336):   c2::u32 coded.bitrate.value = 64000
D/CCodecConfig( 6336):   c2::u32 input.buffers.max-size.value = 65536
D/CCodecConfig( 6336):   c2::u32 input.delay.value = 0
D/CCodecConfig( 6336):   string input.media-type.value = "audio/raw"
D/CCodecConfig( 6336):   string output.media-type.value = "audio/raw"
D/CCodecConfig( 6336):   c2::u32 raw.channel-count.value = 2
D/CCodecConfig( 6336):   c2::u32 raw.pcm-encoding.value = 0
D/CCodecConfig( 6336):   c2::u32 raw.sample-rate.value = 44100
D/CCodecConfig( 6336): }
D/CCodec  ( 6336): [c2.android.raw.decoder] buffers are bound to CCodec for this session
D/CCodecConfig( 6336): no c2 equivalents for durationUs
D/CCodecConfig( 6336): no c2 equivalents for track-id
D/CCodecConfig( 6336): no c2 equivalents for bits-per-sample
D/CCodecConfig( 6336): no c2 equivalents for channel-mask
D/CCodecConfig( 6336): no c2 equivalents for flags
D/CCodecConfig( 6336): c2 config diff is   c2::u32 raw.channel-count.value = 1
D/CCodecConfig( 6336):   c2::u32 raw.sample-rate.value = 8000
W/Codec2Client( 6336): query -- param skipped: index = 1107298332.
D/CCodec  ( 6336): setup formats input: AMessage(what = 0x00000000) = {
D/CCodec  ( 6336):   int32_t channel-count = 1
D/CCodec  ( 6336):   int32_t max-input-size = 65536
D/CCodec  ( 6336):   string mime = "audio/raw"
D/CCodec  ( 6336):   int32_t pcm-encoding = 2
D/CCodec  ( 6336):   int32_t sample-rate = 8000
D/CCodec  ( 6336): } and output: AMessage(what = 0x00000000) = {
D/CCodec  ( 6336):   int32_t channel-count = 1
D/CCodec  ( 6336):   string mime = "audio/raw"
D/CCodec  ( 6336):   int32_t pcm-encoding = 2
D/CCodec  ( 6336):   int32_t sample-rate = 8000
D/CCodec  ( 6336):   int32_t channel-mask = 0
D/CCodec  ( 6336): }
W/Codec2Client( 6336): query -- param skipped: index = 1342179345.
W/Codec2Client( 6336): query -- param skipped: index = 2415921170.
E/FMQ     ( 6336): grantorIdx must be less than 3
E/FMQ     ( 6336): grantorIdx must be less than 3
D/CCodecBufferChannel( 6336): [c2.android.raw.decoder#272] Created input block pool with allocatorID 16 => poolID 17 - OK (0)
I/CCodecBufferChannel( 6336): [c2.android.raw.decoder#272] Created output block pool with allocatorID 16 => poolID 30 - OK
D/CCodecBufferChannel( 6336): [c2.android.raw.decoder#272] Configured output block pool ids 30 => OK
E/ion     ( 6336): ioctl c0044901 failed with code -1: Inappropriate ioctl for device
E/FMQ     ( 6336): grantorIdx must be less than 3
D/CCodecBufferChannel( 6336): [c2.android.raw.decoder#272] MediaCodec discarded an unknown buffer
D/CCodecBufferChannel( 6336): [c2.android.raw.decoder#272] MediaCodec discarded an unknown buffer
I/hw-BpHwBinder( 6336): onLastStrongRef automatically unlinking death recipients
D/BufferPoolAccessor2.0( 6336): bufferpool2 0xeecd0478 : 0(0 size) total buffers - 0(0 size) used buffers - 0/5 (recycle/alloc) - 1/2 (fetch/transfer)
D/BufferPoolAccessor2.0( 6336): evictor expired: 1, evicted: 1
```
</details>

I could trace this back to the call of `TTSHelper.init()` in main.dart:
https://github.com/blockbasti/just_another_workout_timer/blob/b75196114ff9bcfb967ab48c34db4c6936616ce2/lib/main.dart#L21 When I remove the call I reach the home_page as usual.

@omichaelo which Android version are you using?

**Comment by mschmidm** [comment]
This issue might be caused by not having TTS installed (https://github.com/dlutton/flutter_tts/issues/172).

**Comment by blockbasti** [comment]
> On Android 11 (with Google Play) I can not reproduce the issue.

By disabling the "Google Text-to-Speech Engine" app i can reproduce the bug on Android 11. 

I'm currently checking the flutter_tts lib to see if i can fix the issue. If that won't work, we need to find away to detect a missing TTS engine until the upstream bug is fixed.

**Comment by omichaelo** [comment]
> @omichaelo which Android version are you using?

I am on Android 10, using Lineageos. 

> @omichaelo do you have a TTS engine installed?

Nope. Lineageos does not come with a TTS engine.

I got it to work by installing a TTS engine. I used Flite TTS for Android ([Github](https://github.com/happyalu/Flite-TTS-Engine-for-Android) | [F-Droid](https://f-droid.org/en/packages/edu.cmu.cs.speech.tts.flite/)). After selecting it in System Settings the problem is fixed. 

Thank you all! I love this app.
